django/db/models/query.py

```diff
@@ -338,6 +338,20 @@ def __or__(self, other):
         combined.query.combine(other.query, sql.OR)
         return combined
 
+    ##############
+    # DECORATORS #
+    ##############
+
+    def _supported_operation(func):
+        """Decorator function to check supported operation after combinators"""
+        @wraps(func)
+        def wrapped(self, *args, **kwargs):
+            if self.query.combinator:
+                raise NotSupportedError('Calling %s() is not supported after %s().' %
+                                        (func.__name__, self.query.combinator))
+            return func(self, *args, **kwargs)
+        return wrapped
```

Comment:
Sorry for the previous suggestion but decorator looks weird to me :smile: Method would be better, e.g.
```python
def _not_support_combined_queries(self, name):
    if self.query.combinator:
        raise NotSupportedError(
            'Calling %s() is not supported after %s().' % (name, self.query.combinator)
        )
```